lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/FirstFitSchedulingPolicy.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.driver;
+
+import org.apache.reef.tang.annotations.Parameter;
+
+import javax.inject.Inject;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+
+/**
+ * Always select the next worker that has enough resource in a round-robin fashion
+ * based on the worker capacity configured.
+ *
+ * Concurrency: shares the lock with `RunningWorkers'.
+ */
+class FirstFitSchedulingPolicy implements SchedulingPolicy {
+
+  private final int workerCapacity;
+
+  /**
+   * A linked list for a circular buffer of <worker-id, load> pairs.
+   */
+  private final List<Map.Entry<String, Integer>> loadList = new ArrayList<>();
+
+  /**
+   * A list isn't very efficient for fast lookup of worker ids, and so we keep a separate index.
+   */
+  private final HashMap<String, Integer> indexMap = new HashMap<>();
```

Comment:
@johnyangk
Oh, this is a very important assumption that I used without confirming it. Sorry about that, and thanks for pointing it out.

In this case, I guess we should probably go for the simplest implementation, e.g., using a single array for the Random policy, and a single linked list for the First-fit. What do you think?